programs/ssl/ssl_server2.c

```diff
@@ -2737,6 +2747,24 @@ int main( int argc, char *argv[] )
                 mbedtls_ssl_ticket_write,
                 mbedtls_ssl_ticket_parse,
                 &ticket_ctx );
+
+        /* exercise manual ticket rotation (not required for typical use)
+         * (used for external synchronization of session ticket encryption keys)
+         */
+        if( opt.ticket_rotate ) {
+            #define MAX_KEY_BYTES 32    /* 256 bits *//* library/ssl_ticket.c */
```

Comment:
I'm not sure if defining macro inside if statement is consistent with the codding standards. At least for readability, a preprocessor directive should begin in the first column.

Maybe this can be changed to:
`const int max_key_bytes = 32;`

Reply:
Actually, I think this points to a larger problem: other users of this function are likely to need the same constants, and having everyone look at the implementation in `ssl_ticket.c` is not ideal.

Now I'm thinking we should probably export macros `MBEDTLS_SSL_TICKET_MAX_KEY_BYTES` and `MBEDTLS_SSL_TICKET_KEY_NAME_BYTES` in `ssl_ticket.h` (and use them in `ssl_ticket.c` instead of the currently-internal versions).

Should have noticed earlier, a comment that references a `.c` file is often a sign that the public API or documentation is not good enough. (That's another reason why writing tests is useful, helps notice this kind of thing.)